### PR TITLE
[wasm] Enough scafold to enable AOT work with the SDKs

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by kumpera** [comment]
Fixed comments from @jeromelaban

**Comment by kumpera** [comment]
Blocked on #10153

**Comment by kumpera** [comment]
@monojenkins build OS X i386

File: mono/mini/objects.cs [hunk 1]
```diff
@@ -1389,6 +1389,7 @@ sealed class Bar2 : Foo2 {
 		return 1.4e-45f;
 	}
 
+#if !NO_BITCODE
 	[Category ("!BITCODE")] // bug #59953
```
Comment (vargaz) [inline mono/mini/objects.cs]: Why is this disabled if it already has a !BITCODE category ?

Comment (kumpera) [reply mono/mini/objects.cs]: It was crashing the AOT compiler, I'll remove it once its fixed.

**Pull request merged**